pkg/operator/csidriveroperator/olmremovalcontroller.go

```diff
@@ -0,0 +1,463 @@
+package csidriveroperator
+
+import (
+	"context"
+	"time"
+
+	operatorapi "github.com/openshift/api/operator/v1"
+	"github.com/openshift/cluster-storage-operator/pkg/csoclients"
+	"github.com/openshift/cluster-storage-operator/pkg/operator/csidriveroperator/csioperatorclient"
+	"github.com/openshift/cluster-storage-operator/pkg/operatorclient"
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+	"k8s.io/apimachinery/pkg/api/errors"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+	appslisters "k8s.io/client-go/listers/apps/v1"
+	corelisters "k8s.io/client-go/listers/core/v1"
+	"k8s.io/klog/v2"
+)
+
+// This OLMOperatorRemovalController deletes pre-existing CSI driver installed by
+// OLM to new CSI driver namespace.
+// Steps performed:
+// 1. Remove old OLM Subscription and CSV. Remember the operator namespace
+//    (in Storage CR annotation), just in case the controller is restarted
+//    after Subscription removal.
+// 2. Remove the CSI driver Deployment and DaemonSet and wait until they're
+//    really removed. Keep the old namespace around, as it can contain
+//    Manila secrets that are used by existing PVs!
+// 3. Remove the old CR (incl. removing of all of its finalizers).
+// It produces following conditions:
+// <CSI driver name>OLMOperatorRemovalProgressing/Degraded: for status reporting.
```

Comment:
I think this controller is not reporting degraded currently

Reply:
Added